src/ZiraLink.Api.Application/Services/IdentityService.cs

```diff
@@ -0,0 +1,113 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Http;
+using System.Text;
+using System.Text.Json;
+using System.Text.Json.Nodes;
+using System.Threading.Tasks;
+
+using IdentityModel.Client;
+
+using Microsoft.Extensions.Configuration;
+
+using ZiraLink.Api.Application.Enums;
+using ZiraLink.Api.Application.Framework;
+
+namespace ZiraLink.Api.Application.Services
+{
+
+    public class IdentityService : IIdentityService
+    {
+        private readonly IHttpClientFactory _httpClientFactory;
+        private readonly IConfiguration _configuration;
+        private readonly Uri _idsUri; 
+        public IdentityService(IHttpClientFactory httpClientFactory, IConfiguration configuration)
+        {
+            _httpClientFactory = httpClientFactory;
+            _configuration = configuration;
+
+            _idsUri = new Uri(configuration["ZIRALINK_URL_IDS"]!); 
+        }
+         
+
+        public async Task<HttpClient> InitializeHttpClientAsync(CancellationToken cancellationToken)
```

Comment:
The initialize method seems to be used nowhere except in this file, it could be a private method.

Reply:
Done.